### PR TITLE
feature: react error boundary component for pages

### LINKED ISSUE #1824: feat: ErrorBoundary for react based page
Add an ErrorBoundary component:
- catch errors during React rendering 
- display an error fallback page

### PR DESCRIPTION
**Pull request by yomybaby**
resolve #1824 

With this <BAIErrorBoundary/>, you can see the screen below when React rendering encounters an error.

Error examples:
- A request under the suspense feature has failed.
- A not defined variable is used.

If you want to create an error for testing, insert the code below into a React-based component such as `Information`.

```
//@ts-ignore
  notdefinedvariable;
```

### Summary
🚨🛠️🎨

This pull request improves the error handling and code quality of the web UI. It adds the `react-error-boundary` package and a custom `BAIErrorBoundary` component to catch and handle errors in various components. It also removes unused variables and imports from `UserSettingModal.tsx`.

> _To handle errors in React_
> _We added a package that's compact_
> _`BAIErrorBoundary`_
> _Wraps components soundly_
> _And shows a fallback UI that's apt_

### Walkthrough
*  Add `react-error-boundary` package as a dependency to handle errors in React components ([link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-5a3f058f37c9436d08fa97bd30209f67c9a1a61d041b0ce27e48613dfb3a1a10R22))
*  Define a custom `BAIErrorBoundary` component that uses `ErrorBoundary` from the package and displays a fallback UI with a warning message, a reload button, and optionally a reset button and an error message in development mode ([link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-5760b9333632deeff1d62185d14d1ce2ea8a516871f229ce5dbfc0559ac8758eR1-R68))
*  Import and use `BAIErrorBoundary` component in `index.tsx` to wrap the main app components: `Information`, `Routes`, and `StorageHostSettingPage` ([link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-8e8a93d52a7e6e8a4b91770c2de7cb6b500148e2855dc2f5bca6f97f8271bc06R1), [link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-8e8a93d52a7e6e8a4b91770c2de7cb6b500148e2855dc2f5bca6f97f8271bc06L50-R53), [link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-8e8a93d52a7e6e8a4b91770c2de7cb6b500148e2855dc2f5bca6f97f8271bc06L72-R80), [link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-8e8a93d52a7e6e8a4b91770c2de7cb6b500148e2855dc2f5bca6f97f8271bc06L95-R105))
*  Remove unused imports and variables from `UserSettingModal.tsx`: `theme` and `token` ([link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-e38a5a7ebd1f9b1e9548f116319e54f25adaa8df503ec5ac553d3e8fdf53da8fL13-R13), [link](https://github.com/lablup/backend.ai-webui/pull/1890/files?diff=unified&w=0#diff-e38a5a7ebd1f9b1e9548f116319e54f25adaa8df503ec5ac553d3e8fdf53da8fL50))

### CODE REVIEW

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]
image
Can you make backgroundColor of Alert component transparent?

File: react/src/components/BAIErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+const BAIErrorBoundary: React.FC<BAIErrorBoundaryProps> = ({ ...props }) => {
+  return (
+    <ErrorBoundary
+      {...props}
+      fallbackRender={({ error, resetErrorBoundary }) => {
+        console.error('BAIErrorBoundary', error);
+        return (
+          <Result
+            status="warning"
+            title="There is a problem with your request."
```
Comment (agatha197) [inline react/src/components/BAIErrorBoundary.tsx]: How about using `useTranslation` ?

File: react/src/components/BAIErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                          resetErrorBoundary();
+                        }}
+                      >
+                        Reset Error Boundary
+                      </Button>
+                      <Typography.Text>{error.message}</Typography.Text>
+                    </Flex>
+                  }
+                  message={
+                    'This error block is displayed only in WebUI development environment.'
```
Comment (agatha197) [inline react/src/components/BAIErrorBoundary.tsx]: same as above.

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]

**Comment by yomybaby** [comment]
@agatha197  I have pushed some changes you requested. Please review them again and merge
image

**Review by agatha197 — APPROVED** [review]
LGTM!

**Pull request merged**